### ISSUE TITLE
Use systemd socket activation for the docker service

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#9232]
Currently we are always starting the docker daemon, partly because we are using docker-machine to create the VM.

Then we shut down the docker daemon again, when selecting another runtime like containerd or perhaps even cri-o.

Instead we should **not** start up automatically on boot, but only when someone uses the `/var/run/docker.sock`...

In the case of podman/cri-o, this could very well be "never". Also for containerd, it might not "ever" need to use docker ?

----

Since libmachine is dead upstream (deprecated), and we have forked all the drivers, we don't have to care about "API".

So we can provision a machine without actually starting anything on port 2376 (which is also deprecated, see #9229)

By not hardcoding docker everywhere and by stop using the dockershim (in favor of CRI), it makes it better for others.

The podman deployment will use a similar setup, while upgrading from varlink `io.podman.sock` to `podman.sock` (#9230)

http://0pointer.de/blog/projects/socket-activated-containers.html

This does require systemd, but it only applies to **our** OS images.

**Comment by medyagh** [comment]
I agree and this would be a great performance boost for non-docker runtimes

**Comment by afbjorklund** [comment]
The main missing feature (from dockerd) seems to be an idle timeout ?

When running podman for instance, it shuts down again after 5 seconds.

**Comment by afbjorklund** [comment]
More like a roadmap issue for next year, along with the other changes

**Comment by afbjorklund** [comment]
Nobody does this, the docker daemon and containerd (or crio) are _always_ started on boot...

But will continue to use socket-activation for optional components like podman or buildkitd.
